### PR TITLE
feat: organisation creation 

### PR DESCRIPTION
**Pull request by Heba-WebDev**
## What does the PR do?
This PR is about creating a new organisation 

## How should this be manually tested?
**Send a post request to `/api/v1/organisations` with a request body as below**
```json
  {
    "name": "string",
    "description": "string",
    "email": "afq@exmaple.com",
    "industry": "string",
    "type": "string",
    "country": "string",
    "address": "string",
    "state": "string"
  }
```

## Checklist of what you did
- [x] Implemented endpoint to create an organisation
- [x] Validated request body and returned validation errors when validation fails
- [x] Prevented duplicate record by making email unique
- [x] Created an organisation once user is authenticated
- [x] Created a record for the organisation in the database
- [x] Returned success message with response payload
- [x] Added unit tests for organisation service

## Reference Issue
[[FEAT]: API Endpoint For Organisation Creation by users](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/102)

### CODE REVIEW

**Review by TheCodeGhinux — APPROVED** [review]

**Comment by davidolanrewaju** [comment]
Please add documentation

File: src/modules/organisations/organisations.service.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export class OrganisationsService {
+  constructor(
+    @InjectRepository(Organisation)
+    private readonly organisationRepository: Repository<Organisation>,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>
+  ) {}
+
+  async create(createOrganisationDto: OrganisationRequestDto, userId: string) {
+    const emailFound = await this.emailExsits(createOrganisationDto.email);
```
Comment (coderoyalty) [inline src/modules/organisations/organisations.service.ts:20]: It looks like the method name `emailExsits` is misspelt. Did you mean to write: `emailExists`?

Comment (Heba-WebDev) [reply src/modules/organisations/organisations.service.ts:20]: Yes, it's a typo. I'll fix this right now @coderoyalty

Comment (Cobidex) [inline src/modules/organisations/organisations.service.ts]: Just semantics... Change this to emailExists. Overall the file looks good to me

**Comment by Heba-WebDev** [comment]
@davidolanrewaju Ismael added swagger docs globally, that's why I didn't add it.

**Review by Cobidex — APPROVED** [review]

**Review by rugue — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by buka4rill — APPROVED** [review]
LGTM

**Pull request merged**